support ftp-urls in WebClient

alexlehm opened the pull request:
add same url handling to WebClient for ftp-urls as was added to vertx-core in the pr https://github.com/eclipse/vert.x/pull/1959

the code probably needs some improvements, e.g. constructing the url with a URL constructor instead of creating the String directly

tsegismont commented on vertx-web-client/src/main/java/io/vertx/ext/web/client/impl/HttpRequestImpl.java:
```diff
@@ -278,7 +287,19 @@ private void send(String contentType, Object body, Handler<AsyncResult<HttpRespo
+          URI uri = new URI(protocol, null, host, port, requestURI, null, null);
+          req = client.requestAbs(method, uri.toString());
+        } catch (URISyntaxException ex) {
+          // FIXME: what is proper error handling?
```
This looks like the right way to handle the error to me.

tsegismont commented on vertx-web-client/src/test/java/WebClientTest.java:
```diff
@@ -0,0 +1,45 @@
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+
+/**
+ * 
```
Remove ?

alexlehm commented:
@tsegismont actually test unit test isn't really useful as it relies on Squid running on the same machine, if we want to test it, we need a fake proxy as in core http tests

tsegismont commented:
IIRC the core build produces a jar with tests. Is the fake proxy an
internal class? Could it be made reusable?

2017-05-16 22:48 GMT+02:00 Alexander Lehmann <notifications@github.com>:

> @tsegismont <https://github.com/tsegismont> actually test unit test isn't
> really useful as it relies on Squid running on the same machine, if we want
> to test it, we need a fake proxy as in core http tests
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/vert-x3/vertx-web/pull/625#issuecomment-301911101>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABbltskAkpcmvFRphj0tUiR97iSVHisRks5r6gumgaJpZM4NYogO>
> .
>

alexlehm commented:
turns out that is already available as WebClientTest extends HttpTestBase from vertx-core
I have changed the test to use the test proxy and added it to the test class

tsegismont commented:
LGTM. Thanks @alexlehm
